plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -57,7 +57,7 @@
      * Client settings derived from those in {@link #staticClientSettings} by combining them with settings
      * in the {@link RepositoryMetadata}.
      */
-    private volatile Map<S3ClientSettings, Map<RepositoryMetadata, S3ClientSettings>> derivedClientSettings = emptyMap();
+    private volatile Map<Settings, S3ClientSettings> derivedClientSettings = emptyMap();
```

Comment:
This was way over-engineered. Since we don't have any selective cache invalidation when the static settings get refreshed, using this kind of complicated map was completely pointless.